matchzoo/models/duet.py

```diff
@@ -0,0 +1,230 @@
+                              "the distributed model."))
+        params.add(Param(name='dm_right_pool_size', value=100,
+                         desc="Kernel size of 1D convolution layer in "
+                              "the distributed model."))
+        params.add(Param(
+            name='dropout_rate', value=0.5,
+            hyper_space=hyper_spaces.quniform(low=0.0, high=0.8, q=0.02),
+            desc="The dropout rate."))
+        return params
+
+    @classmethod
+    def get_default_preprocessor(
+        cls,
+        truncated_mode: str = 'pre',
+        truncated_length_left: int = 30,
+        truncated_length_right: int = 30,
+        filter_mode: str = 'df',
+        filter_low_freq: float = 1,
+        filter_high_freq: float = float('inf'),
+        remove_stop_words: bool = False,
+        ngram_size: int = 3
+    ):
+        """:return: Default preprocessor."""
+        return preprocessors.BasicPreprocessor(
+            truncated_mode=truncated_mode,
+            truncated_length_left=truncated_length_left,
+            truncated_length_right=truncated_length_right,
+            filter_mode=filter_mode,
+            filter_low_freq=filter_low_freq,
+            filter_high_freq=filter_high_freq,
+            remove_stop_words=remove_stop_words,
+            ngram_size=ngram_size
+        )
+
+    @classmethod
+    def _xor_match(cls, x):
+        """Xor match of two inputs."""
+        t1 = x[0]
+        t2 = x[1]
+        t1_shape = list(t1.shape)
+        t2_shape = list(t2.shape)
+        t1_expand = torch.unsqueeze(t1, 2).repeat(1, 1, t2_shape[1])
+        t2_expand = torch.unsqueeze(t2, 1).repeat(1, t1_shape[1], 1)
+        out = torch.eq(t1_expand, t2_expand).float()
+        return out
+
+    def build(self):
+        """Build model structure."""
+        self.lm_conv1d = nn.Conv1d(
+            in_channels=self._params['right_length'],
+            out_channels=self.params['lm_filters'],
+            kernel_size=1,
+            stride=1
+        )
+        lm_mlp_size = self._params['left_length'] * self._params['lm_filters']
+        self.lm_mlp = self._make_multi_layer_perceptron_layer(lm_mlp_size)
+        self.lm_linear = self._make_perceptron_layer(
+            in_features=self._params['mlp_num_fan_out'],
+            out_features=1
+        )
+
+        self.dm_conv_activation_func = parse_activation(
+                self._params['dm_conv_activation_func']
+        )
+        self.dm_conv_left = nn.Conv1d(self._params['vocab_size'],
+                                      self._params['dm_filters'],
+                                      self._params['dm_kernel_size']
+                                      )
+        self.dm_mlp_left = self._make_perceptron_layer(
+            in_features=self._params['dm_filters'],
+            out_features=self._params['dm_filters']
+        )
+        self.dm_conv1_right = nn.Conv1d(self._params['vocab_size'],
```

Comment:
Code indent like this may be better.
```suggestion
        self.dm_conv1_right = nn.Conv1d(
            self._params['vocab_size'],
            self._params['dm_filters'],
            self._params['dm_kernel_size']
        )
```